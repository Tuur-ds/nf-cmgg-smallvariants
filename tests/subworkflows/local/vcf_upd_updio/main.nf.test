nextflow_workflow {

    name "Test Workflow VCF_UPD_UPDIO"
    script "subworkflows/local/vcf_upd_updio/main.nf"
    workflow "VCF_UPD_UPDIO"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_upd_updio"

    test("vcf_upd_updio - default - family") {

        when {
            params {
                updio = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"Ashkenazim", family:"Ashkenazim", family_count:3, caller:"haplotypecaller"],
                    file(params.famvcf, checkIfExists:true),
                    file(params.famtbi, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"Ashkenazim", family:"Ashkenazim", family_count:3, caller:"haplotypecaller"],
                    file(params.ped, checkIfExists:true)
                ])
                input[2] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.updio
                ).match("default - family") }
            )
        }

    }

    test("vcf_upd_updio - default - sample") {
        // This should output nothing
        when {
            params {
                updio = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"Ashkenazim", family:"Ashkenazim", family_count:1, caller:"haplotypecaller"],
                    file(params.vcf1, checkIfExists:true),
                    file(params.tbi1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"Ashkenazim", family:"Ashkenazim", family_count:1, caller:"haplotypecaller"],
                    file(params.ped, checkIfExists:true)
                ])
                input[2] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.updio
                ).match("default - sample") }
            )
        }

    }

}
