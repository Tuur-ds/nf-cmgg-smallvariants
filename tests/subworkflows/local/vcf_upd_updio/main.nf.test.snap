{
    "default - family": {
        "content": [
            [
                [
                    {
                        "id": "Ashkenazim",
                        "family": "Ashkenazim",
                        "family_count": 3,
                        "caller": "haplotypecaller",
                        "child": "NA24385",
                        "mother": "NA24143",
                        "father": "NA24149"
                    },
                    [
                        "NA24385.events_list:md5,c61379c7a4bb61cbe9612b5cda773cf4",
                        "NA24385.log:md5,a40a24f379127a9cde7e40a1ce1032ec",
                        "NA24385.table:md5,ca8165fc7869a113ca034396de7cf579",
                        "NA24385.upd:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-27T13:36:54.518507627"
    },
    "default - sample": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-27T13:37:01.70605131"
    }
}